lib/mixlib/install/product.rb

```diff
@@ -20,17 +20,21 @@
 module Mixlib
   class Install
     class Product
-      def initialize(&block)
+      def initialize(key, &block)
+        @product_key = key
         instance_eval(&block)
       end
 
       DSL_PROPERTIES = [
         :config_file,
         :ctl_command,
+        :product_key,
```

Comment:
We wanted to the key to be available in an instance of `Mixlib::Install::Product`.